Sources/Quick/DSL/World+DSL.swift

```diff
@@ -19,6 +19,9 @@ extension World {
     }
 
     internal func describe(description: String, flags: FilterFlags, closure: () -> ()) {
+        guard currentExampleMetadata == nil else {
+            raiseError("'describe' cannot be used inside 'it', 'describe' may only be used inside 'context' or 'describe'. ")
```

Comment:
> My only concern here is that cases like the following won't print the correct diagnostic:

You're totally right. That's a detail that I missed. Thanks for pointing it out!

> Perhaps we could add a method to look at the newly-introduced properties on the ExampleGroup and produce a string giving the name of the currently-executing block. Does that make any sense?

Yes, it does. It would allow the code to do something *similar* to what that validator-type class in original PR tried to accomplish.